datafusion/core/src/execution/context/mod.rs

```diff
@@ -2247,6 +2292,196 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn register_deregister_udf() -> Result<()> {
```

Comment:
Would it be possible to move these tests to https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/user_defined/mod.rs 

Doing so would make them more discoverable as well as likely shorter as you could probably avoid having to define ScalarUDFs